tests/Validator.test.js

```diff
@@ -79,93 +79,107 @@ describe('Validator', function () {
+        invalid: ['foobaradsdd','adalova..com','next..test.com'],
+        invalidMessage: ['Must be a URL']
+      },
+      date: {
+        validatorTerm: Validator.date(),
+        valid: ['January 9, 2013','December 20, 1903'],
+        invalid: ['foo', 'Jan 2012 1202'],
+        invalidMessage: ['Must be a date']
+      },
+      before: {
+        validatorTerm: Validator.before('January 10, 2010'),
+        valid: ['January 9 2010'],
+        invalid: ['January 9 2011'],
+        invalidMessage: ['Must be before January 10, 2010']
+      },
+      after: {
+        validatorTerm: Validator.after('January 10, 2010'),
+        valid: ['January 15 2010'],
+        invalid: ['February 8 2000'],
+        invalidMessage: ['Must be after January 10, 2010']
+      },
+      alpha: {
+        validatorTerm: Validator.alpha(),
+        valid: ['aasdDSAasd','la','Ada'],
+        invalid: ['ad1asdd','Ada1ovelace'],
+        invalidMessage: ['Must be letters only (A-Z)']
+      },
+      email: {
+        validatorTerm: Validator.email(),
+        valid: ['aad@dasd.com'],
+        invalid: ['asdd.com','ada@lovelace..com','bar@@foo.com'],
+        invalidMessage: ['Must be an email']
+      },
+      creditCard: {
+        validatorTerm: Validator.creditCard(),
+        valid: ['4012888888881881'],
+        invalid: ['1231231232','11111888888881881333'],
+        invalidMessage: ['Please enter a valid credit card']
+      },
+      max: {
+        validatorTerm: Validator.max(10),
+        valid: [9],
+        invalid: [11],
+        invalidMessage: ['Must be less than 10']
+      },
+      min: {
+        validatorTerm: Validator.min(10),
+        valid: ['10'],
+        invalid: ['8'],
+        invalidMessage: ['Must be greater than 10']
+      },
+      maxLength: {
+        validatorTerm: Validator.maxLength(5),
+        valid: ['abcde'],
+        invalid: ['asdadsadasda'],
+        invalidMessage: ['Must be less than 5 characters']
+      },
+      minLength: {
+        validatorTerm: Validator.minLength(2),
+        valid: ['adasdasdasd'],
+        invalid: ['a'],
+        invalidMessage: ['Must be at least 2 characters']
+      },
+      pattern: {
+        validatorTerm: Validator.pattern(/foo/),
+        valid: ['foo'],
+        invalid: ['bar'],
+        invalidMessage: ['Does not match pattern']
+      },
+      currency: {
+        validatorTerm: Validator.currency(),
+        valid: ['42.42','2897.99','2829873'],
+        invalid: ['42..42','..38','32.3'],
+        invalidMessage: ['Must be a valid currency']
+      },
     };
 
     Object.keys(testData).forEach(key => {
       var data = testData[key];
-      it('#' + key, function () {
-        should.deepEqual(data[0].assert(data[1][0]), data[1][1]);
-        should.deepEqual(data[0].assert(data[2][0]), data[2][1]);
+      it('#' + key, () => {
+        data.valid.forEach( e => {
```

Comment:
It would be a better idea to name this something descriptive, such as `testCase`